mindsdb/integrations/handlers/duckdb_handler/duckdb_handler.py

```diff
@@ -0,0 +1,184 @@
+    def connect(self) -> DuckDBPyConnection:
+        """Connect to a DuckDB database.
+
+        Returns:
+            DuckDBPyConnection: The database connection.
+        """
+
+        if self.is_connected is True:
+            return self.connection
+
+        self.connection = duckdb.connect(self.connection_data['database'])
+        self.is_connected = True
+
+        return self.connection
+
+    def disconnect(self):
+        """Close the database connection."""
+
+        if self.is_connected is False:
+            return
+
+        self.connection.close()
+        self.is_connected = False
+
+    def check_connection(self) -> StatusResponse:
+        """Check the connection to the DuckDB database.
+
+        Returns:
+            StatusResponse: Connection success status and error message if an error occurs.
+        """
+
+        response = StatusResponse(False)
+        need_to_close = self.is_connected is False
+
+        try:
+            self.connect()
+            response.success = True
+        except Exception as e:
+            log.logger.error(
+                f'Error connecting to DuckDB {self.connection_data["database"]}, {e}!'
+            )
+            response.error_message = str(e)
+        finally:
+            if response.success is True and need_to_close:
+                self.disconnect()
+            if response.success is False and self.is_connected is True:
+                self.is_connected = False
+
+        return response
+
+    def native_query(self, query: str) -> Response:
+        """Execute a SQL query.
+
+        Args:
+            query (str): The SQL query to execute.
+
+        Returns:
+            Response: The query result.
+        """
+        need_to_close = self.is_connected is False
+
+        connection = self.connect()
+        cursor = connection.cursor()
+
+        try:
+            cursor.execute(query)
+
+            result = cursor.fetchall()
+            if result:
+                response = Response(
+                    RESPONSE_TYPE.TABLE,
+                    data_frame=pd.DataFrame(
+                        result, columns=[x[0] for x in cursor.description]
+                    ),
+                )
+            else:
+                connection.commit()
+                response = Response(RESPONSE_TYPE.OK)
+        except Exception as e:
+            log.logger.error(
+                f'Error running query: {query} on {self.connection_data["database"]}!'
+            )
+            response = Response(RESPONSE_TYPE.ERROR, error_message=str(e))
+
+        cursor.close()
+        if need_to_close is True:
+            self.disconnect()
+
+        return response
+
+    def query(self, query: ASTNode) -> Response:
+        """Render and execute a SQL query.
+
+        Args:
+            query (ASTNode): The SQL query.
+
+        Returns:
+            Response: The query result.
+        """
+
+        if isinstance(query, ASTNode):
+            query_str = query.to_string()
```

Comment:
Since duckdb is PostgreSQL dialect wise we should use it here. For e.g check the PostgreSQL handler and re use the same renderer https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/postgres_handler/postgres_handler.py#L132